src/components/funFacts/FunFactComponent.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import { funFacts,  FunFact  as FunFactInterface } from "./funfacts";
+import "./FunFactPage.scss";
+
+interface FunFactProps {
+    funFacts: FunFactInterface[];
+}
+
+export const FunFact: React.FunctionComponent<FunFactProps> = ({ funFacts }) => {
+
+  const [selectedFactId, setSelectedFactId] = useState(0);
+
+  const toggleFactDetail = (funFactId: number) => {
+    if (selectedFactId === funFactId){
+      setSelectedFactId(0);
+    }
+    else {
+      setSelectedFactId(funFactId);
+    }
+  }
+
+  return(
+    <div>
+      {funFacts.map(funFact => (
+        <div className="funfact">
+          <div className="funfact__prompt">
+            <h2 className="funfact__prompt-question">{funFact.prompt}</h2>
+            <button className={`funfact__button ${funFact.id===selectedFactId ? "funfact__button--revealed" : ""}`} onClick={ () => toggleFactDetail(funFact.id)}>^</button>
```

Comment:
as above
```suggestion
            <button className={`fun-fact__button ${funFact.id===selectedFactId ? "fun-fact__button--revealed" : ""}`} onClick={ () => toggleFactDetail(funFact.id)}>^</button>
```

Reply:
Also this line is really long. Could you add some line breaks throughout?

Reply:
Also what is the `^` character doing at the end of the line?
```suggestion
            <button className={`funfact__button ${funFact.id===selectedFactId ? "funfact__button--revealed" : ""}`} onClick={ () => toggleFactDetail(funFact.id)}></button>
```